database/dbProgramInterestForm.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+function createProgramInterestForm($form) {
+    $connection = connect();
+
+    $family_id = $_SESSION['_id'];
+    $first_name = $form["first_name"];
+    $last_name = $form["last_name"];
+    $address = $form["address"];
+	$city = $form["city"];
+    $neighborhood = $form["neighborhood"];
+    $state = $form["state"];
+    $zip = $form["zip"];
+    $cell_phone = $form["cell_phone"];
+    $home_phone = $form["home_phone"];
+    $email = $form["email"];
+    $child_num = $form["child_num"];
+    $child_ages = $form["child_ages"];
+    $adult_num = $form["adult_num"];
+	$id = null;
+
+    // Begin transaction so if there are any errors with any queries, they are all rolled back and no data is inserted into database
+    // This is done so if one query is succesful and another isn't, it doesn't just insert the successful one which would result in incomplete data in the database
+    mysqli_begin_transaction($connection);
+    try {
+        // First query inserts basic information in dbProgramInterestForm
+        $query = "
+            INSERT INTO dbProgramInterestForm (family_id, first_name, last_name, address, city, neighborhood, state, zip, cell_phone, home_phone, email, child_num, 
+            child_ages, adult_num)
+            values ('$family_id', '$first_name', '$last_name', '$address', '$city', '$neighborhood', '$state', '$zip', '$cell_phone', '$home_phone', '$email', '$child_num', 
+            '$child_ages', '$adult_num');
+        ";
+        $result = mysqli_query($connection, $query);
+        if (!$result) {
+            return null;
+        }
+        $id = mysqli_insert_id($connection);
+
+        // Insert program interests
+        if (isset($form['programs'])) {
+            $programs = $form['programs'];
+            insertProgramInterests($programs, $id, $connection);
+        }
+        // Insert topic interests
+        if (isset($form['topics'])) {
+            $topics = $form['topics'];
+            insertTopicInterests($topics, $id, $connection);
+        }
+        // insert availability
+        $days = $form['days'];
+        insertAvailability($days, $id, $connection);
+        // Commit transaction if there were no problems inserting data
+        mysqli_commit($connection);
+    } catch (Exception $e) {
+        // If there was a problem inserting data roll back and return null
+        mysqli_rollback($connection);
+        return null;
+    }
+    mysqli_close($connection);
+    return $id;
+}
+
+function insertProgramInterests($programs, $form_id, $connection) {
+    // Second query inserts ids in dbProgramInterestsForm_ProgramInterests junction table
+    $query = "INSERT INTO dbProgramInterestsForm_ProgramInterests (form_id, interest_id) values ";
```

Comment:
idk if I'm reading this wrong but I don't see where interest_id value comes from here. if it automatically creates the id here ignore this